espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/LocaleHelpers.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.espressoserver.lib.helpers
+
+import android.content.Context
+import android.os.Build
+import android.os.LocaleList
+
+import io.appium.espressoserver.lib.model.LocaleParams
+import java.util.*
+
+fun changeLocale(context: Context, locale: Locale) {
+    if (Build.VERSION.SDK_INT >= 26) {
+        Locale.setDefault(Locale.Category.DISPLAY, locale)
+    } else {
+        Locale.setDefault(locale)
+    }
+    val res = context.resources
+    val config = res.configuration
+    config.locales = LocaleList(locale)
```

Comment:
`config.setLayoutDirection(locale)` is necessary to apply RTL style properly

Reply:
The docstring on setLocales method:

```
    /**
     * Set the locale list. This is the preferred way for setting up the locales (instead of using
     * the direct accessor or {@link #setLocale(Locale)}). This will also set the layout direction
     * according to the first locale in the list.
     *
     * Note that the layout direction will always come from the first locale in the locale list,
     * even if the locale is not supported by the resources (the resources may only support
     * another locale further down the list which has a different direction).
     *
     * @param locales The locale list. If null, an empty LocaleList will be assigned.
     */
```

Reply:
ah, okay. I see.
I also understand this is application-wide change (via the target context), so other areas such as status bar and other applications do not change.
Will check with an app which has multiple locale/layout later

Reply:
LocaleList is only for API level 24

Reply:
thanks, will change